### PR TITLE
docs: document DnD related part names

### PR DESCRIPTION
**Pull request by FrediWa**
This PR adds documentatinon to drag-and-drop related part names.

### CODE REVIEW

File: articles/components/grid/styling.adoc [hunk 1]
```diff
@@ -288,9 +288,28 @@ Cell in row with open details:: `vaadin-grid+++<wbr>+++**::part(details-open-row
 Row with open details:: `vaadin-grid+++<wbr>+++**::part(details-open-row)**`
 
 
-
-
-
+=== Drag and Drop
```
Comment (jouni) [inline articles/components/grid/styling.adoc:291]:
```suggestion
=== Drag & Drop
```

File: articles/components/grid/styling.adoc [hunk 2]
```diff
@@ -288,9 +288,28 @@ Cell in row with open details:: `vaadin-grid+++<wbr>+++**::part(details-open-row
...
+The following terminology is used in the description of these part names:
+
+- Ghost: the semi-transparent copy of the row that is dragged around by the pointer
+- Drag-source: the non-moving rows that are being dragged
+- Drop-target: a row or a gap between rows that the ghost can be dropped onto
```
Comment (jouni) [inline articles/components/grid/styling.adoc:296]:
```suggestion
Drag source:: The row that is being dragged by the user. This part doesn't visually move during the drag.

Ghost:: A copy of the drag source (i.e., dragged row), which follows the user's pointer during the drag operation.

Drop target:: A row or a space between rows that the ghost can be dropped onto.
```

File: articles/components/grid/styling.adoc [hunk 3]
```diff
@@ -288,9 +288,28 @@ Cell in row with open details:: `vaadin-grid+++<wbr>+++**::part(details-open-row
...
+
+&#10;
```
Comment (jouni) [inline articles/components/grid/styling.adoc:298]: I assume this is here to separate the two lists. Let’s use what the AsciiDoc documentation suggests: https://docs.asciidoctor.org/asciidoc/latest/lists/separating/

```suggestion
// separate lists
```

File: articles/components/grid/styling.adoc [hunk 4]
```diff
@@ -288,9 +288,28 @@ Cell in row with open details:: `vaadin-grid+++<wbr>+++**::part(details-open-row
...
+Cell in a row that ghost hovers above, with drop-target below:: `vaadin-grid+++<wbr>+++**::part(dragover-below-row-cell)**`
+Cell in a drop-target row that ghost hovers above:: `vaadin-grid+++<wbr>+++**::part(dragover-on-top-row-cell)**`
+Cell in a row that isn’t draggable:: `vaadin-grid+++<wbr>+++**::part(drag-disabled-row-cell)**`
+Cell in a row that isn’t a drop-target:: `vaadin-grid+++<wbr>+++**::part(drop-disabled-row-cell)**`
+Row being dragged (the “ghost”):: `vaadin-grid+++<wbr>+++**::part(dragstart-row)**`
+Row that ghost hovers above, with drop-target above:: `vaadin-grid+++<wbr>+++**::part(dragover-above-row)**`
+Row that ghost hovers above, with drop-target below:: `vaadin-grid+++<wbr>+++**::part(dragover-below-row)**`
+Drop-target row that ghost hovers above:: `vaadin-grid+++<wbr>+++**::part(dragover-on-top-row)**`
+A row that isn’t draggable:: `vaadin-grid+++<wbr>+++**::part(drag-disabled-row)**`
+A row that isn’t a drop-target :: `vaadin-grid+++<wbr>+++**::part(drop-disabled-row)**`
```
Comment (jouni) [inline articles/components/grid/styling.adoc:312]: I tried to make this more understandable, but I’m not sure I succeeded.

```suggestion
The cells in the “ghost” row::
`vaadin-grid+++<wbr>+++**::part(dragstart-row-cell)**`

The cells in the drag source row::
`vaadin-grid+++<wbr>+++**::part(drag-source-row-cell)**`

The cells of a row when the drop target is between this row and the row before it (i.e., above this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-above-row-cell)**`

The cells of a row when the drop target is between this row and the row after it (i.e., below this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-below-row-cell)**`

The cells of a row when the drop target is the row itself (i.e., on top of this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-on-top-row-cell)**`

The cells of a row that isn’t draggable::
`vaadin-grid+++<wbr>+++**::part(drag-disabled-row-cell)**`

The cells of a row that isn’t a drop target::
`vaadin-grid+++<wbr>+++**::part(drop-disabled-row-cell)**`

The dragged row (i.e., the “ghost”)::
`vaadin-grid+++<wbr>+++**::part(dragstart-row)**`

The row which is the drop target, and the target is between this row and the row before it (i.e., above this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-above-row)**`

The row which is the drop target, and the target is between this row and the row after it (i.e., below this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-below-row)**`

The row which is the drop target, and the target is the row itself (i.e., on top of this row)::
`vaadin-grid+++<wbr>+++**::part(dragover-on-top-row)**`

A row that isn’t draggable::
`vaadin-grid+++<wbr>+++**::part(drag-disabled-row)**`

A row that isn’t a drop target ::
`vaadin-grid+++<wbr>+++**::part(drop-disabled-row)**`
```

File: articles/components/grid/styling.adoc (hunk 2 above)
Comment (FrediWa) [reply articles/components/grid/styling.adoc:296]: I'm not sure it's a good idea to make the lists appear the same as they convey very different information. The explanations aren't part names so they should not appear as such.

File: articles/components/grid/styling.adoc (hunk 4 above)
Comment (FrediWa) [reply articles/components/grid/styling.adoc:312]: Yeah, the part names are hard to describe,. This PR is the result of Rolf's feedback put on top of my (someone fairly new to the grid) understanding of the parts.

File: articles/components/grid/styling.adoc (hunk 2 above)
Comment (jouni) [reply articles/components/grid/styling.adoc:296]: That’s a good point. I changed the suggestion back to a regular list, but made the terms bold for emphasis.

File: articles/components/grid/styling.adoc (hunk 1 above)
Comment (FrediWa) [reply articles/components/grid/styling.adoc]: Done

File: articles/components/grid/styling.adoc (hunk 3 above)
Comment (FrediWa) [reply articles/components/grid/styling.adoc]: Done

**Review by jouni — APPROVED** [review]

**Pull request merged**